src/client/QXmppAccountMigrationManager.cpp

```diff
@@ -161,7 +159,7 @@ struct QXmppAccountMigrationManagerPrivate {
         std::function<QXmppTask<Result<std::any>>()> exportFunction;
     };
 
-    std::unordered_map<std::type_index, ExtensionData> extensions;
+    std::map<std::type_index, ExtensionData> extensions;
```

Comment:
Using `map` instead of `unordered_map` should be avoided if possible. `map` is much slower than `unordered_map`

![image](https://github.com/user-attachments/assets/bea3c2cd-fdf1-49da-9661-d2398433bca1)

unordered_map does search, insert and delete in average O(1) and worst case O(log(n)), while map always takes O(log(n)) which makes a huge difference with large `n`.

Please rather fix the comparison in the unit tests.

Reply:
I'm well aware about the difference between map and unordered_map and the consequences of it.
This map will host at best ~20 entries, it's not as if it would make any difference at such low number. We would have to have 10k/15k at a minimum so that the difference starts to impact something.
Anyway, I don't know how I would allow std::unordered_map to have a stable seed/hash for unit tests, if you have idea or something i can do for that just tell me and I will happily revert/apply, else i will simply keep moving to map because as of now there is no downside to move to.

Reply:
I came up with a partial solution, see last commit and tell me if it's sufficient for you.